src/main/scala/io/iohk/ethereum/ledger/Ledger.scala

```diff
@@ -15,26 +15,122 @@ import scala.annotation.tailrec
 
 trait Ledger {
 
-  def executeBlock(block: Block, validators: Validators): Either[BlockExecutionError, Seq[Receipt]]
+  def executeBlock(block: Block): Either[BlockExecutionError, Seq[Receipt]]
 
-  def prepareBlock(block: Block, validators: Validators): BlockPreparationResult
+  def prepareBlock(block: Block): BlockPreparationResult
 
   def simulateTransaction(stx: SignedTransaction, blockHeader: BlockHeader): TxResult
+
+  def importBlock(block: Block): BlockImportResult
+
+  def resolveBranch(headers: Seq[BlockHeader]): BranchResolutionResult
 }
 
 //FIXME: Make Ledger independent of BlockchainImpl, for which it should become independent of WorldStateProxy type
-class LedgerImpl(vm: VM, blockchain: BlockchainImpl, blockchainConfig: BlockchainConfig) extends Ledger with Logger {
+class LedgerImpl(vm: VM, blockchain: BlockchainImpl, blockchainConfig: BlockchainConfig, validators: Validators) extends Ledger with Logger {
```

Comment:
This is up to discussion, but maybe we could separate the block execution and branch resolution aspects of the Ledger into separate traits?